js/main.js

```diff
@@ -404,39 +404,56 @@ function initiate() {
             );
         }
 
+        function addAllTypicalWarningsInFragment(fragment, fragmentIndex, badness, errorCode, type) {
+            var badPos = 0;
+            var offset = 0;
+            var subFragment = fragment.substring(0);
+            while (true) {
+                badPos = subFragment.search(badness);
+                if (badPos >= 0) {
+                    addTypicalWarning(errorCode, type, fragmentIndex, offset + badPos);
+                    var matched = subFragment.match(badness)[0];
+                    offset += badPos + matched.length;
+                    subFragment = subFragment.substring(badPos + matched.length);
+                } else {
+                    break;
+                }
+            }
+        }
+
         function addWarningQuick(fragmentType, badness, errorCode, mathFragmentType) {
             if(fragmentType == 'math') {
                 for (var i = 0; i < mathFragments.length; ++i) {
                     if( mathFragmentType && mathFragmentType != mathFragmentTypes[i] ) {
                         continue;
                     }
-                    var badPos = mathFragments[i].search(badness);
-                    if (badPos >= 0){
-                        addTypicalWarning(errorCode, 'math', i, badPos);
-                    }
+                    addAllTypicalWarningsInFragment(mathFragments[i], i, badness, errorCode, 'math');
                 }
             }
             else {
                 for (var i = 0; i < textFragments.length; ++i) {
-                    var badPos = textFragments[i].search(badness);
-                    if (badPos >= 0){
-                        addTypicalWarning(errorCode, 'text', i, badPos);
-                    }
+                    addAllTypicalWarningsInFragment(textFragments[i], i, badness, errorCode, 'text');
                 }
             }
         }
 
-        /* STAGE: Check if there are no teacher fixmes left */
-        var badPos = latexString.search(/\\fix\{/);
-        if (badPos >= 0){
-            addWarning('FIXME_NOT_YET_FIXED', null, '\\fix{', latexString.substring(0,badPos).split('\n').length-1);
+        function addAllWarningsLatexString(errorCode, errorRegex) {
+            if (!errorRegex.flags.includes("g")) {
+                var errorRegexString = errorRegex.toString().substring(1, errorRegex.toString().length - 1);
+                errorRegex = new RegExp(errorRegexString, "g");
```

Comment:
Wait, no, that's not how it works. Say, I have a regex `var re = /abc/`, without the global flag. I also have a string `var s = "abc abc abc"`. No matter how many times I run `re.exec(s)`, it will only return the first match, at index 0. If, however, I have a regex `var re = /abc/g`, with the global flag, every time I run `re.exec(s)`, it will return consecutive matches.